### PR TITLE
Add kubernetes monitoring role

### PR DESCRIPTION
**Pull request by matofeder**
Related PRs
- https://github.com/osism/ansible-playbooks/pull/420
- https://github.com/osism/defaults/pull/181

SCS issue: https://github.com/SovereignCloudStack/issues/issues/576

### CODE REVIEW

File: roles/kubernetes_monitoring/defaults/main.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+# operator
+
+operator_user: dragon
+operator_group: "{{ operator_user }}"
+
+##########################
+# kubernetes_monitoring
+
+kubernetes_monitoring_chart_source: "dnationcloud/dnation-kubernetes-monitoring-stack"
+kubernetes_monitoring_chart_version: "3.5.0"
```
Comment (berendt) [inline roles/kubernetes_monitoring/defaults/main.yml:12]: Add `# renovate datasource=github-releases depName= dnationcloud/dnation-kubernetes-monitoring-stack)` in the line before this line.

File: roles/kubernetes_monitoring/defaults/main.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+kubernetes_monitoring_template_directory: .
+## kubernetes namespace in which the operator should be installed
+kubernetes_monitoring_namespace: kubernetes-monitoring
+
+##########################
+# openstack_exporter
+
+enable_openstack_exporter: false
+openstack_exporter_chart_source: "oci://registry.scs.community/openstack-exporter/prometheus-openstack-exporter"
+openstack_exporter_chart_version: "0.4.5"
```
Comment (berendt) [inline roles/kubernetes_monitoring/defaults/main.yml:26]: Add a renovate line before this line. Probably `# renovate: datasource=docker depName=registry.scs.community/openstack-exporter/prometheus-openstack-exporter`

File: roles/kubernetes_monitoring/defaults/main.yml (hunk 1 above)
Comment (matofeder) [reply roles/kubernetes_monitoring/defaults/main.yml:13]: see 15b597fe13a1c91db5b87a72f90c44fe681698b2

File: roles/kubernetes_monitoring/defaults/main.yml (hunk 2 above)
Comment (matofeder) [reply roles/kubernetes_monitoring/defaults/main.yml:28]: see 15b597fe13a1c91db5b87a72f90c44fe681698b2
We can take a GitHub release as a version source also in this case

**Review by artificial-intelligence — CHANGES_REQUESTED** [review]

File: roles/kubernetes_monitoring/README.rst [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+**Kubernetes monitoring Variables**
+
+.. zuul:rolevar:: kubernetes_monitoring_chart_version
+   :default: 3.5.0
+
+Version of Kubernetes monitoring helm chart version which should be used.
+
+.. zuul:rolevar:: kubernetes_monitoring_work_directory
+   :default: /tmp/kubernetes_monitoring/configuration
```
[the remark this answers is not in the document]
Comment (berendt) [reply roles/kubernetes_monitoring/README.rst:28]: I think that's ok. The configuration is generated temporarily and then transferred to Kubernetes.

File: roles/kubernetes_monitoring/defaults/main.yml [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+## credentials for the clouds.yml configured in the secure.yml
+openstack_exporter_secure_yml_path: /opt/configuration/environments/openstack/secure.yml
+
+
+## TEST
+openstack_cacert: /etc/ssl/certs/ca-certificates.crt
+kolla_internal_vip_address: 192.168.16.9
+kolla_external_vip_address: 192.168.16.254
+kolla_internal_fqdn: api-int.testbed.osism.xyz
+kolla_external_fqdn: api.testbed.osism.xyz
```
[the remark this answers is not in the document]
Comment (berendt) [reply roles/kubernetes_monitoring/defaults/main.yml]: That's ok. We have the same in other places.

**Comment by berendt** [comment]
Merging here. Will be transferred to osism/osism-kubernetes.

**Pull request merged**